docs/src/references/c8y-firmware-management.md

```diff
@@ -0,0 +1,214 @@
+        C8Y Firmware Plugin->>Tedge Agent: Symlink the cached firmware file to file-transfer repository and generate a `tedge.url` for it
+        C8Y Firmware Plugin->>Child Device Connector: MQTT: firmware_update request to `child-id` with `tedge.url`
+
+        Child Device Connector ->> C8Y Firmware Plugin: MQTT: firmware_update response with operation status: "executing" 
+        C8Y Firmware Plugin ->> C8Y Cloud: MQTT: Update c8y_Firmware operation status to "EXECUTING"
+        Child Device Connector->>Tedge Agent: HTTP: Download the firmware file from `tedge.url`
+        Child Device Connector->>Child Device Connector: Apply downloaded firmware file
+        Child Device Connector ->> C8Y Firmware Plugin: MQTT: firmware_update response with operation status: "successful" 
+
+        C8Y Firmware Plugin ->> C8Y Cloud: MQTT: Update c8y_Firmware operation status to "SUCCESSFUL"
+        C8Y Firmware Plugin ->> C8Y Firmware Plugin: Remove the symlink from file-transfer repository but keep the cached firmware copy for reuse
+```
+
+The following keywords are used in the following section for brevity:
+
+* `TEDGE_DATA_PATH`: The path set by tedge config `data.path`. Default: `/var/tedge`
+* `TEDGE_TMP_PATH`: The path set by tedge config `tmp.path`. Default: `/tmp`
+* `FIRMWARE_CACHE_PATH`: `$TEDGE_DATA_PATH/cache`
+* `FIRMWARE_OP_PATH`: `$TEDGE_DATA_PATH/firmware`
+* `FILE_TRANSFER_REPO`: `$TEDGE_DATA_PATH/file-transfer`
+* `TEDGE_HTTP_ADDRESS`: The combination of tedge configs `http.address`:`http.port`
+* `OP_ID`: An operation ID
+* `FILE_ID`: A firmware file id derived from the SHA-256 digest of the firmware url
+
+1. The plugin, on recept of a `c8y_Firmware` request from Cumulocity for a child device named `$CHILD_DEVICE_ID`
+   in the format `515,$CHILD_DEVICE_ID,$FIRMWARE_NAME,$FIRMWARE_VERSION,$FIRMWARE_URL`
+   1. Validate if the same firmware update operation is already in progress
+      by iterating over all the operation files in the `$FIRMWARE_OP_PATH` directory.
+      The operation files contains the last `firmware_update` request's JSON payload along with the `device` ID.
+      If an operation file with the `device` id, `name`, `version` and `url` fields matching
+      the incoming `$FIRMWARE_NAME`,`$FIRMWARE_VERSION` and `$FIRMWARE_URL` is found,
+      the same request is re-sent to the child device by just incrementing the `attempt` count value.
+      The operation file content is also overwritten the with updated `attempt` count. 
+   1. If a pending operation match is not found, do a look up if the firmware file for the given url already exists
+      in its firmware cache at `$FIRMWARE_CACHE_PATH`.
+      The file name for the lookup is derived from the SHA-256 digest of the firmware url.
+      If a cached copy exists, the download is skipped.
+   1. If not found in the firmware cache, the plugin downloads the firmware file from the `url` to `$TEDGE_TMP_PATH`.
```

Comment:
After we changed `Downloader` implementation, the plugin (to be precise downloader) downloads the file to the final destination directory (so here `$TEDGE_DATA_PATH/cache`) with the name `<name>.tmp`. 

Here's downloader code: https://github.com/thin-edge/thin-edge.io/blob/main/crates/common/download/src/download.rs#L302

Actually we should remove loading `tmp` path from firmware-plugin...